Fix detection of AR and RANLIB

The configure.in script assumes that ranlib and ar are necessarily
prefixed by ${ac_tool_prefix}, which is the value of --host. However,
it's not necessarily the case.

So instead, use AC_CHECK_TOOL to check for AR, and AC_PROG_RANLIB to
check for RANLIB.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: b/configure.in
===================================================================
--- a/configure.in
+++ b/configure.in
@@ -621,10 +621,8 @@
 BUILDDIR=$(pwd)
 AC_SUBST(BUILDDIR)
 
-AR=${ac_tool_prefix}ar
-AC_SUBST([AR])
-RANLIB=${ac_tool_prefix}ranlib
-AC_SUBST([RANLIB])
+AC_CHECK_TOOL([AR], [ar])
+AC_PROG_RANLIB
 
 dnl =======================================================================
 dnl Output our results.
